solr/core/src/test/org/apache/solr/handler/admin/MetricsHandlerTest.java

```diff
@@ -308,19 +318,24 @@ public void testKeyMetrics() throws Exception {
     handler.handleRequestBody(req(CommonParams.QT, "/admin/metrics", CommonParams.WT, "json",
         MetricsHandler.KEY_PARAM, key1, MetricsHandler.KEY_PARAM, key2, MetricsHandler.KEY_PARAM, key3), resp);
     values = resp.getValues();
-    val = values.findRecursive("metrics", key1);
+    map = new HashMap<>();
+    values.toMap(map);
+    val = Utils.getObjectByPath(map, false, "metrics/" + key1);
     assertNotNull(val);
-    val = values.findRecursive("metrics", key2);
+    val = Utils.getObjectByPath(map, true, "metrics/" + key2);
     assertNotNull(val);
-    val = values.findRecursive("metrics", key3);
+    val = Utils.getObjectByPath(map, true, "metrics/" + key3);
     assertNotNull(val);
 
     String key4 = "solr.core.collection1:QUERY./select.requestTimes:1minRate";
     resp = new SolrQueryResponse();
     handler.handleRequestBody(req(CommonParams.QT, "/admin/metrics", CommonParams.WT, "json",
         MetricsHandler.KEY_PARAM, key4), resp);
     values = resp.getValues();
-    val = values.findRecursive("metrics", key4);
+    map = new HashMap<>();
```

Comment:
There is no need to convert this to `Map`.

`values._get()` can fetch nested values